src/mpd.c

```diff
@@ -3725,23 +3737,39 @@ mpd_command_toggleoutput(struct evbuffer *evbuf, int argc, char **argv, char **e
  * Example output:
  *   outputid: 0
  *   outputname: Computer
+ *   plugin: alsa
  *   outputenabled: 1
  *   outputvolume: 50
+ * https://mpd.readthedocs.io/en/latest/protocol.html#audio-output-devices
  */
 static void
 speaker_enum_cb(struct player_speaker_info *spk, void *arg)
 {
-  struct evbuffer *evbuf;
+  struct output_outputs_param *param = arg;
+  struct evbuffer *evbuf = param->buf;
+  char plugin[sizeof(spk->output_type) + 1];
+  char *p;
+  char *q;
 
-  evbuf = (struct evbuffer *)arg;
+  /* MPD outputs lowercase plugin (audio_output:type) so convert to
+   * lowercase, convert spaces to underscores to make it a single word */
+  for (p = spk->output_type, q = plugin; *p != '\0'; p++, q++)
+    {
+      *q = tolower(*p);
+      if (*q == ' ')
+      	*q = '_';
+    }
+  *q = '\0';
 
   evbuffer_add_printf(evbuf,
-		      "outputid: %d\n"
+		      "outputid: %u\n"
 		      "outputname: %s\n"
+		      "plugin: %s\n"
 		      "outputenabled: %d\n"
 		      "outputvolume: %d\n",
-		      (unsigned short) spk->id,
+		      param->nextid++,
```

Comment:
Like above, put the increase in a separate line, makes it easier to read and more consistent with the rest